tools/check_resource_attestation.py

```
@@ -0,0 +1,577 @@
+        return reject("attestation:policy-rollback")
+
+    # --- manifest structural checks ---
+    if not isinstance(manifest, dict):
+        return reject("manifest:non-canonical")
+    mac = manifest.get("mac_attestation")
+    ledger = manifest.get("ledger_attestation")
+    if not isinstance(mac, dict):
+        return reject("manifest:duplicated-attestation" if isinstance(mac, list) else "manifest:ledger-only-not-final")
+    if not isinstance(ledger, dict):
+        return reject("manifest:duplicated-attestation" if isinstance(ledger, list) else "manifest:mac-only-not-final")
+    for role, att in (("mac", mac), ("ledger", ledger)):
```

Ruff (0.15.21)

[warning] Loop control variable `role` not used within loop body

Rename unused `role` to `_role`

(B007)